### PR TITLE
parsing lables of override-account/subaccount-id

### PR DESCRIPTION
**Pull request by lauraanddolay**
implement [Kyma configurability of specific runtimes](https://github.tools.sap/kyma/backlog/issues/2683), now there are three types :
```
1.   overrides-account-{plan label}: true
2.   overrides-account-{global account ID}: true
3.   overrides-subaccount-{subaccount ID}: true
```
Note: If overrides attributes are the same (e.g `application-operator.controller.resources.limits.cpu` )with differ values , the latter one (e.g `subaccount`)will replace the previous one (e.g. `account`) in the [AppendOverrides()](https://github.com/kyma-project/control-plane/blob/main/components/kyma-environment-broker/internal/process/input/input.go#L144)

### CODE REVIEW

File: components/kyma-environment-broker/internal/runtimeoverrides/runtimeoverrides.go [hunk 1]
```diff
@@ -99,35 +117,48 @@ func (ro *runtimeOverrides) collectFromSecrets() (map[string][]*gqlschema.Config
...
+
+		if err := ro.k8sClient.List(ro.ctx, configMaps, listOpts...); err != nil {
+			errMsg := fmt.Sprintf("cannot fetch list of config maps: %s", err)
+			return componentsOverrides, globalOverrides, errors.Wrap(err, errMsg)
+		}
+
+		for _, cm := range configMaps.Items {
+			component, global := getComponent(cm.Labels)
+			for key, value := range cm.Data {
+				ro.log.Infof("collectFromConfigMaps() component , global: %s %s overrideType key value : %s %s %s", component, global, overrideType, key, value)
```
Comment (ebensom) [inline components/kyma-environment-broker/internal/runtimeoverrides/runtimeoverrides.go:147]:
```suggestion
				ro.log.Debugf("collectFromConfigMaps() component , global: %s %s overrideType key value : %s %s %s", component, global, overrideType, key, value)
```

File: components/kyma-environment-broker/internal/runtimeoverrides/runtimeoverrides.go [hunk 2]
```diff
@@ -99,35 +117,48 @@ func (ro *runtimeOverrides) collectFromSecrets() (map[string][]*gqlschema.Config
...
+	for _, k := range keys {
+		overrideType := OverridesMapping[k]
+		ro.log.Infof("collectFromConfigMaps() overrideType %s on account %s subaccount %s\n", overrideType, account, subaccount)
+		configMaps := &coreV1.ConfigMapList{}
+		listOpts := configMapListOptions(overrideType, overrideList[k], overridesVersion)
+
+		if err := ro.k8sClient.List(ro.ctx, configMaps, listOpts...); err != nil {
+			errMsg := fmt.Sprintf("cannot fetch list of config maps: %s", err)
+			return componentsOverrides, globalOverrides, errors.Wrap(err, errMsg)
+		}
```
[the remark this answers is not in the document]
Comment (lauraanddolay) [reply components/kyma-environment-broker/internal/runtimeoverrides/runtimeoverrides.go:142]: Hi @ebensom , for the ` version+globalaccount and version+subaccount overrides is optional` , i tested with below two cases:
1. [testcase 1st t.Run() ](https://github.com/kyma-project/control-plane/blob/main/components/kyma-environment-broker/internal/runtimeoverrides/runtimeoverrides_test.go#L18) 
2. on kcp-dev, without adding the configmap of `vergsion+globalaccount/subaccount`,  do provisioning skr on image-PR1781 (also chekcing the [client.go#L258](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/client.go#L258) ---> [metadata_client.go#L145](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/metadata_client.go#L145)), its err is nil and the loop continue.

I found that the below code will return `err == nil` and its print out `len(configMaps.Items) =0`, i think that we may not need to consider the `!apierrors.IsNotFound(err)` condition and just reuse (like original error code) . Is that acceptable, pls correct me if i am wrong.
```
if err := ro.k8sClient.List(ro.ctx, configMaps, listOpts...); err != nil && (strings.HasPrefix(labelKey) || !apierrors.IsNotFound(err)) {
		errMsg := fmt.Sprintf("cannot fetch list of config maps: %s", err)
		return componentsOverrides, globalOverrides, errors.Wrap(err, errMsg)
}
...
fmt.Println(len(configMaps.Items))
```

File: components/kyma-environment-broker/internal/process/upgrade_kyma/secret_cm_overrides.go [hunk 3]
```diff
@@ -38,22 +39,32 @@ func NewOverridesFromSecretsAndConfigStep(os storage.Operations, runtimeOverride
...
 		return s.operationManager.OperationFailed(operation, "invalid operation provisioning parameters", nil, log)
 	}
 
+	globalAccountID := operation.ProvisioningParameters.ErsContext.GlobalAccountID
+	subAccountID := operation.ProvisioningParameters.ErsContext.SubAccountID
+	if globalAccountID == "" || subAccountID == "" {
+		log.Errorf("cannot find global accountID '%s' or subAccountID '%s' ", globalAccountID, subAccountID)
+		return s.operationManager.OperationFailed(operation, "invalid operation provisioning parameters on globalAccount/subAccount", nil, log)
+	}
+
```
[the remark this answers is not in the document]
Comment (lauraanddolay) [reply components/kyma-environment-broker/internal/process/upgrade_kyma/secret_cm_overrides.go:59]: Get it and I'll remove it.

File: components/kyma-environment-broker/internal/runtimeoverrides/runtimeoverrides.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (lauraanddolay) [reply components/kyma-environment-broker/internal/runtimeoverrides/runtimeoverrides.go]: follow the comments and updated now.

File: components/kyma-environment-broker/internal/runtimeoverrides/runtimeoverrides.go [hunk 4]
```diff
@@ -6,20 +6,27 @@ import (
...
+	namespace                      = "kcp-system"
+	componentNameLabel             = "component"
+	overridesVersionLabelPrefix    = "overrides-version-"
+	overridesPlanLabelPrefix       = "overrides-plan-"
+	overridesAccountLabelPrefix    = "overrides-account-"
+	overridesSubaccountLabelPrefix = "overrides-subaccount-"
+	overridesSecretLabel           = "runtime-override"
+	PLANNAME                       = "planeName"
+	ACCOUNT                        = "account"
+	SUBACCOUNT                     = "subaccount"
```
Comment (ebensom) [inline components/kyma-environment-broker/internal/runtimeoverrides/runtimeoverrides.go:29]: I think these are not needed any more, not used

**Review by ebensom — COMMENTED** [review]
LGTM, asking review from @piotrmiskiewicz

**Comment by ebensom** [comment]
@lauraanddolay also don't forget to bump the KEB image version in resources/kcp/values.yaml

File: docs/kyma-environment-broker/03-06-runtime-overrides.md [hunk 5]
```diff
@@ -6,15 +6,19 @@ You can set overrides to customize your Kyma Runtime. To provision a cluster wit
 
 ## ConfigMap
 
-The overrides mechanism selects ConfigMaps by filtering the resources using labels. You can prepare overrides for a given plan and Kyma version using the `overrides-plan-{PLAN_NAME}: "true"` and `overrides-version-{KYMA_VERSION}: "true"` labels.
+The overrides mechanism selects ConfigMaps by filtering the resources using labels. You can prepare overrides for three kinds of combination:
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-06-runtime-overrides.md:9]:
```suggestion
The overrides mechanism selects ConfigMaps by filtering the resources using labels. You can prepare overrides for three kinds of combinations:
```

File: docs/kyma-environment-broker/03-06-runtime-overrides.md [hunk 6]
```diff
@@ -6,15 +6,19 @@ You can set overrides to customize your Kyma Runtime. To provision a cluster wit
...
+1. a given plan and Kyma version using the `overrides-plan-{PLAN_NAME}: "true"` and `overrides-version-{KYMA_VERSION}: "true"` labels.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-06-runtime-overrides.md:10]:
```suggestion
- A given plan and a Kyma version using the `overrides-plan-{PLAN_NAME}: "true"` and `overrides-version-{KYMA_VERSION}: "true"` labels.
```

File: docs/kyma-environment-broker/03-06-runtime-overrides.md [hunk 7]
```diff
@@ -6,15 +6,19 @@ You can set overrides to customize your Kyma Runtime. To provision a cluster wit
...
+2. a given global account ID and Kyma version using the `overrides-account-{GLOBAL_ACCOUNT_ID}: "true"` and `overrides-version-{KYMA_VERSION}: "true"` labels.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-06-runtime-overrides.md:11]:
```suggestion
- A given global account ID and a Kyma version using the `overrides-account-{GLOBAL_ACCOUNT_ID}: "true"` and `overrides-version-{KYMA_VERSION}: "true"` labels.
```

File: docs/kyma-environment-broker/03-06-runtime-overrides.md [hunk 8]
```diff
@@ -6,15 +6,19 @@ You can set overrides to customize your Kyma Runtime. To provision a cluster wit
...
+3. a given subaccount ID and Kyma version using the  `overrides-account-{SUB_ACCOUNT_ID}: "true"` and `overrides-version-{KYMA_VERSION}: "true"` labels.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-06-runtime-overrides.md:12]:
```suggestion
- A given subaccount ID and a Kyma version using the  `overrides-account-{SUB_ACCOUNT_ID}: "true"` and `overrides-version-{KYMA_VERSION}: "true"` labels.
```

File: docs/kyma-environment-broker/03-06-runtime-overrides.md [hunk 9]
```diff
@@ -6,15 +6,19 @@ You can set overrides to customize your Kyma Runtime. To provision a cluster wit
...
 
 > **NOTE:** Each ConfigMap that defines overrides must have both labels assigned.
 
 Optionally, you can narrow the scope of the overrides to a specific component. Use the `component: "{COMPONENT_NAME}"` label to indicate the component.
 
 The overrides lookup mechanism requires at least one ConfigMap present for each plan and version pair. Otherwise, it fails.
 
-See the example of a ConfigMap with global overrides for the `trial` plan and versions `1.15.1` and `1.16.0`:
+See the examples of a ConfigMap：
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-06-runtime-overrides.md:20]:
```suggestion
See the examples of different ConfigMaps:
```

File: docs/kyma-environment-broker/03-06-runtime-overrides.md [hunk 10]
```diff
@@ -6,15 +6,19 @@ You can set overrides to customize your Kyma Runtime. To provision a cluster wit
...
+1. Configmap with global overrides for the `trial` plan and versions `1.15.1` and `1.16.0`:
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-06-runtime-overrides.md:21]:
```suggestion
- ConfigMap with global overrides for the `trial` plan and versions `1.15.1` and `1.16.0`:
```

File: docs/kyma-environment-broker/03-06-runtime-overrides.md [hunk 11]
```diff
@@ -30,6 +34,36 @@ data:
   global.disableLegacyConnectivity: "true"
 ```
 
+2. Configmap with global overrides for the `global account` ID and versions `1.15.1` and `1.16.0`:
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-06-runtime-overrides.md:37]:
```suggestion
- ConfigMap with global overrides for the global account ID and versions `1.15.1` and `1.16.0`:
```

File: docs/kyma-environment-broker/03-06-runtime-overrides.md [hunk 12]
```diff
@@ -30,6 +34,36 @@ data:
...
+  labels:
+    overrides-account-d9994f8f-7e46-42a8-b2c1-1cddd9d2ef06: "true"
+    overrides-version-1.15.1: "true"
+  name: global-overrides
+  namespace: kcp-system
+data:
+  global.disableLegacyConnectivity: "true"
+```
+
+2. Configmap with global overrides for the `subaccount` ID and versions `1.16.0`:
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-06-runtime-overrides.md:52]:
```suggestion
- ConfigMap with global overrides for the subaccount ID and version `1.16.0`:
```

File: docs/kyma-environment-broker/03-06-runtime-overrides.md (hunk 6 above)
Comment (lauraanddolay) [reply docs/kyma-environment-broker/03-06-runtime-overrides.md]: modified it.

File: docs/kyma-environment-broker/03-06-runtime-overrides.md (hunk 8 above)
Comment (lauraanddolay) [reply docs/kyma-environment-broker/03-06-runtime-overrides.md]: changed it.

File: docs/kyma-environment-broker/03-06-runtime-overrides.md (hunk 10 above)
Comment (lauraanddolay) [reply docs/kyma-environment-broker/03-06-runtime-overrides.md]: Done.

File: docs/kyma-environment-broker/03-06-runtime-overrides.md (hunk 11 above)
Comment (lauraanddolay) [reply docs/kyma-environment-broker/03-06-runtime-overrides.md]: Done.

File: docs/kyma-environment-broker/03-06-runtime-overrides.md (hunk 9 above)
Comment (lauraanddolay) [reply docs/kyma-environment-broker/03-06-runtime-overrides.md]: Fixed.

File: docs/kyma-environment-broker/03-06-runtime-overrides.md (hunk 12 above)
Comment (lauraanddolay) [reply docs/kyma-environment-broker/03-06-runtime-overrides.md]: Done.

**Review by klaudiagrz — APPROVED** [review]

**Comment by lauraanddolay** [comment]
Close it because of requirement of [task2683]( https://github.tools.sap/kyma/backlog/issues/2683) is updated.

**Pull request closed without merging**